packages/features/voting-context.tsx

```diff
@@ -71,19 +109,118 @@ function VotingCtxProvider({ children }: ChildrenProps) {
     },
     [partyList],
   );
+  // end of Election related
+
+  // Voting related
+  const [voteInput, setVoteInput] = useState('');
+  const [isVoting, setIsVoting] = useState(true);
+  const [endMessage, setEndMessage] = useState('FIM');
+  const secretCode = '12345';
+
+  const selectedPartyData = useMemo(() => {
+    return {
+      party: partyList.filter((e) => e.code === voteInput.slice(0, 2))[0],
+      candidate: function () {
+        return {
+          name: this.party?.members.candidate.name,
+          img: this.party?.members.candidate.image,
+        };
+      },
+      vice: function () {
+        return {
+          name: this.party?.members.viceCandidate.name,
+          img: this.party?.members.viceCandidate.image,
+        };
+      },
+      partyInfo: function () {
+        return {
+          code: this.party?.code,
+          name: this.party?.name,
+          slug: this.party?.slug,
+        };
+      },
+    };
+  }, [partyList, voteInput]);
+
+  const updateVoteInput = useCallback(
+    (input: string) => {
+      if (voteInput.length > 5) return;
+
+      if (input.charCodeAt(0) >= 48 && input.charCodeAt(0) <= 57) {
+        setVoteInput(voteInput + input);
+      }
+    },
+    [voteInput],
+  );
+
+  // TODO 'are you sure' step
+
+  const voteBlank = useCallback(() => {
+    handleVote('Branco');
+    console.log('voce votou em branco');
+  }, []);
+
+  const voteConfirm = useCallback(() => {
+    if (voteInput === secretCode) return handleVotingEnd();
+    if (selectedPartyData.party)
+      return handleVote(selectedPartyData.partyInfo().name);
+
+    console.log('selecione um partido valido');
```

Comment:
The behavior for this case is not the one coded. If the user enters an invalid party code the page should display the "NULO" option for the user, letting the user confirm if he desires to cancel his vote.
Check the example below that I've found in this [simulator](https://www.tse.jus.br/hotsites/simulador-de-votacao/depFederal.html):

![image](https://user-images.githubusercontent.com/14007745/191599011-2a69e65e-ef52-4482-a944-5c87a0d0dc7d.png)

We should use this simulator behavior as our base standard.

This task is also kinda huge, maybe we could break this in an enhancement, what do you think?

Reply:
Humm, I see you updated the feature but, when the user adds an invalid party number, nothing on the screen changes, alerting the user that he is nullifying his vote. I believe we should follow the example in the image, displaying "Número errado" and "Voto nulo" to the user.  I think an easy way to achieve that would be to create a simple ternary:

```javascript
{isValidPartyNumber ? <ValidPartyInfo> : <InvalidPartyInfo> }
```
In the `PartyInfo` component.



Also, I've noticed that when I press "Branco" the vote is already computed and my session is terminated, we should add an intermediary state here also. So, when the user click on "Branco", it is shown to him that he choose "Branco"  so he can confirm or fix it.

Reply:
Separado em novas tasks